Jellyfin.Api/Controllers/UserLibraryController.cs

```diff
@@ -206,21 +206,19 @@ public ActionResult<IEnumerable<BaseItemDto>> GetLocalTrailers([FromRoute, Requi
                 : _libraryManager.GetItemById(itemId);
 
             var dtoOptions = new DtoOptions().AddClientFields(Request);
-            var dtosExtras = item.GetExtras(new[] { ExtraType.Trailer })
-                .Select(i => _dtoService.GetBaseItemDto(i, dtoOptions, user, item))
-                .ToArray();
 
             if (item is IHasTrailers hasTrailers)
             {
                 var trailers = hasTrailers.LocalTrailers;
                 var dtosTrailers = _dtoService.GetBaseItemDtos(trailers, dtoOptions, user, item);
-                var allTrailers = new BaseItemDto[dtosExtras.Length + dtosTrailers.Count];
-                dtosExtras.CopyTo(allTrailers, 0);
-                dtosTrailers.CopyTo(allTrailers, dtosExtras.Length);
+                var allTrailers = new BaseItemDto[dtosTrailers.Count];
+                dtosTrailers.CopyTo(allTrailers, 0);
                 return allTrailers;
             }
 
-            return dtosExtras;
+            return item.GetExtras(new[] { ExtraType.Trailer })
+                .Select(i => _dtoService.GetBaseItemDto(i, dtoOptions, user, item))
+                .ToArray();
```

Comment:
```suggestion
            return Ok(item.GetExtras()
                .Where(e => e.ExtraType == ExtraType.Trailer)
                .Select(i => _dtoService.GetBaseItemDto(i, dtoOptions, user, item)));
```

Reply:
Btw, the `GetExtras` overload used previously doesn't sort.
https://github.com/jellyfin/jellyfin/blob/7500c2b28b4603965461b7bce37413c0a53ff2d0/MediaBrowser.Controller/Entities/BaseItem.cs#L2634-L2640

Reply:
Do you think this could be an issue?

Reply:
I don't think so. This is a question of consistency.
Moreover, perhaps we don't need to sort trailers at all, or even we need to randomize them.
Speaking of minimizing the number of operation, we should probably sort where we call `GetExtras` if necessary.